src/irmin/tree_intf.ml

```diff
@@ -227,7 +227,8 @@ module type S = sig
       For every node [n], ui [n] is a leaf node, call [f path n]. Otherwise:
 
       - Call [pre path n]. By default [pre] is the identity;
-      - Recursively call [fold] on each children, in lexicographic order;
+      - Recursively call [fold] on each children - the order in which we visit
+        the children is undefined and unstable;
```

Comment:
Even if `unstable` is true, maybe we can try to be a bit more precise here? Could we provide any guarantee to our users here?

Reply:
is [this](https://github.com/mirage/irmin/pull/1525/files#diff-8c3a3ff44e610c7200cae01bd11f0faa1e003ea21c5e44e73a7ff226cc578935R230-R232) better ?